src/Components/VitalsMonitor/HL7PatientVitalsMonitor.tsx

```diff
@@ -25,6 +27,32 @@ export default function HL7PatientVitalsMonitor(props: IVitalsComponentProps) {
   const { patient, bed, asset } = props.patientAssetBed ?? {};
   const authUser = useAuthUser();
 
+  const [bedAssignmentStartDate, setBedAssignmentStartDate] = useState(
+    new Date()
+  );
+
+  const {
+    res: bedsRes,
+    data: bedsData,
+    refetch: bedsFetch,
+  } = useQuery(routes.listConsultationBeds, {
+    query: {
+      consultation: props.consultationId,
+    },
+  });
+
+  useEffect(() => {
+    if (!bedsData) console.log("error");
+    else {
+      const startDate = new Date(bedsData.results[0].created_date);
+      setBedAssignmentStartDate(startDate);
+    }
+  }, [props.consultationId, bedsRes, bedsData]);
```

Comment:
use the `onResponse` property of `useQuery` instead of doing this in `useEffect`